pkg/images/enricher/enricher.go

```diff
@@ -68,6 +68,10 @@ type EnrichmentContext struct {
 	// Set via ad-hoc requests such as "roxctl image scan".
 	Delegable bool
 
+	// ClusterID contains the ID of the cluster to delegate the enrichment request to if the request is Delegable.
+	// Used to override the delegated registry configuration.
+	ClusterID string
```

Comment:
Wondering how to make a better distinction between this `ClusterID` here and the one contained in the `Source`. They might be within the enricher two completely different code paths, but from a client's POV it's not really clear which of the two to set (and to what effect).

Can we re-use the `Source` instead, or make a sub-struct for delegated stuff?

Reply:
> Wondering how to make a better distinction between this ClusterID here and the one contained in the Source.  They might be within the enricher two completely different code paths, but from a client's POV it's not really clear which of the two to set (and to what effect).

I believe `ClusterID`'s comment clarifies the effect/intent. Would you consider that sufficient?

The `ClusterID` fields are opposites, `Source.ClusterID` represents where the request came from whereas `ClusterID` represents (in this case) the destination.  `Source.ClusterID` is not used currently after the `SourcedAutogeneratedIntegrations` feature was permanently disabled, unclear if it will be enabled in the future. But should it be re-used in the future I can imagine both being in use simultanously.

In the initial design for registry mirroring `ClusterID` would have also represented the cluster context to use when evaluating mirrors - if the cluster was a valid delegation target the request would have been delegated, otherwise the mirrors would have been evaluated in Central using the `ClusterID` - I don't believe we can make the assumption that `ClusterID` will only be used for delegation - which makes me lean toward keeping it as is instead of creating a sub-struct specific for delegated stuff.

All things considered this is an internal struct that is only ever in mem, we can change it with minimal impact if we feel the fields need better organization.